aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp

```diff
@@ -581,8 +588,73 @@ inline static void svd_cusolver_gesvdjBatched(const Tensor& A, const Tensor& U,
   });
 }
 
-// Check if gesvdj results converge. If not, return a vector that contains the non-converging batch index.
-// If the returned vector is empty, all the matrices converged.
+template<typename scalar_t>
+inline static void apply_svd_cusolver_gesvdaStridedBatched(const Tensor& A, const Tensor& U, const Tensor& S, const Tensor& V,
+    const Tensor& infos, bool full_matrices, bool compute_uv) {
+  using value_t = typename c10::scalar_value_type<scalar_t>::type;
+  int m = cuda_int_cast(A.size(-2), "m");
+  int n = cuda_int_cast(A.size(-1), "n");
```

Comment:
Just to be sure, please add a `TORCH_INTERNAL_ASERT(m > n);`
Then, you can just use `n` in place of `k`, `S_stride` and `rank`.